apps/prairielearn/src/pages/course_instance/lookup/lookupByEnrollmentCode.ts

```diff
@@ -19,34 +22,53 @@ router.get(
   '/',
   asyncHandler(async (req, res) => {
     // Parse and validate the code parameter
-    const { code, course_instance_id: courseInstanceId } = LookupCodeSchema.parse(req.query);
+    const { code, course_instance_id: courseInstanceIdToCheck } = LookupCodeSchema.parse(req.query);
 
     // Look up the course instance by enrollment code
-    const courseInstance = await selectOptionalCourseInstanceByEnrollmentCode(code);
-
-    // User-facing terminology is to use "course" instead of "course instance"
-
-    if (!courseInstance) {
+    const courseInstanceId = await selectOptionalCourseInstanceIdByEnrollmentCode({
+      enrollment_code: code,
+    });
+    if (!courseInstanceId) {
+      // User-facing terminology is to use "course" instead of "course instance"
       res.status(404).json({
         error: 'No course found with this enrollment code',
       });
       return;
     }
 
-    if (courseInstanceId && courseInstance.id !== courseInstanceId) {
+    if (courseInstanceIdToCheck && courseInstanceId !== courseInstanceIdToCheck) {
       res.status(404).json({
         error: 'This enrollment code is for a different course',
       });
       return;
```

Comment:
My initial reaction was "hmm, it seems bad to inform the user that an enrollment code is valid for a _different_ course". But I guess this is not really a problem because one could just enter an enrollment code on the homepage to try it out, right?